Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -274,7 +275,21 @@ private static bool IsIonNamedProperty(PropertyInfo property)
         {
             return property.GetCustomAttribute(typeof(IonPropertyName)) != null;
         }
-        
+
+        /// <summary>
+        /// We only serde public, internal, and protected internal properties.
+        /// </summary>
+        private static bool HasValidAccessModifier(PropertyInfo propertyInfo)
+        {
+            var methodInfo = propertyInfo.GetGetMethod(true);
+
+            if (methodInfo is not null)
+            {
+                return methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly;
+            }
+            return false;
```

Comment:
We can improve this even more with `return methodInfo is not null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly);`

Reply:
Done